projects/alembic/alembic_dump_info_fuzzer.cc

```diff
@@ -0,0 +1,206 @@
+  schema.getFaceVaryingPropagateCornersProperty();
+  schema.getInterpolateBoundaryProperty();
+}
+
+void dumpFaceSet(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IFaceSet faceSet(node.getParent(), header.getName());
+  const IFaceSetSchema &schema = faceSet.getSchema();
+  schema.getNumSamples();
+}
+
+void dumpCurves(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const ICurves curves(node.getParent(), header.getName());
+  const ICurvesSchema &schema = curves.getSchema();
+
+  dumpAttributes(schema);
+}
+
+void dumpXform(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  const IXform xform(node.getParent(), header.getName());
+  const IXformSchema &schema = xform.getSchema();
+
+  schema.getNumSamples();
+  schema.getNumOps();
+}
+
+void dumpMaterial(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  IMaterial material(node.getParent(), header.getName());
+  IMaterialSchema &schema = material.getSchema();
+
+  std::vector<std::string> targetNames;
+
+  for (const std::string &target : targetNames) {
+    std::vector<std::string> shaderTypes;
+    schema.getShaderTypesForTarget(target, shaderTypes);
+    const size_t shaderTypeCount = shaderTypes.size();
+    for (size_t s = 0; s < shaderTypeCount; s++) {
+
+      ICompoundProperty parameters =
+          schema.getShaderParameters(target, shaderTypes[s]);
+      const size_t parameterCount = parameters.getNumProperties();
+    }
+  }
+}
+
+void dumpNodes(const IObject &node) {
+  const ObjectHeader &header = node.getHeader();
+  // Dump the general node information.
+  header.getName();
+  header.getFullName();
+  header.getMetaData().serialize();
+
+  // Dump the type specific information.
+  if (Alembic::AbcGeom::IPolyMesh::matches(header)) {
+    dumpPolyMesh(node);
+  } else if (Alembic::AbcGeom::ISubD::matches(header)) {
+    dumpSubD(node);
+  } else if (Alembic::AbcGeom::IFaceSet::matches(header)) {
+    dumpFaceSet(node);
+  } else if (Alembic::AbcGeom::ICurves::matches(header)) {
+    dumpCurves(node);
+  } else if (Alembic::AbcGeom::IXform::matches(header)) {
+    dumpXform(node);
+  } else if (Alembic::AbcMaterial::IMaterial::matches(header)) {
+    dumpMaterial(node);
+  } else { // Miscellaneous nodes such as the root.
+    ;
+  }
+
+  // Dump the child headers.
+  const size_t childCount = node.getNumChildren();
+  for (size_t i = 0; i < childCount; i++) {
+    dumpNodes(node.getChild(i));
+  }
+}
+
+void dumpInfo(const char *file) {
+  // Load the Alembic archive and verify that it is valid.
+  IFactory factory;
+  IArchive archive = factory.getArchive(file);
+
+  if (archive.valid()) {
+    archive.getName();
+    dumpNodes(archive.getTop());
+  }
+}
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  FuzzerTemporaryFile tempFile(data, size);
+  const char *filename = tempFile.filename();
```

Comment:
just
dumpInfo(tempFile.filename());

i think this can not be NULL, makes this nice and short...